### ISSUE TITLE
Description of scripts in keyboard help are not translatable

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#80]
**Reported by mdcurran on 2 Apr 2007 09:39 UTC**
Because descriptions of scripts in NVDA are doc strings, it makes it very hard to allow them to be translatable for various languages. Rather than the doc string being set inside the function, it should be manually set after the function is defined, so that the string can be enclosed in _()

**Comment by michaelDCurran** [comment]
**Comment by mdcurran on 2 Apr 2007 09:42 UTC**
A few scripts have already been changed in appModules/_default.py, though there are  still many to go there, plus in NVDA objects and virtual Buffers.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 17 Apr 2007 22:18 UTC**

**Comment by michaelDCurran** [comment]
**Comment by pvagner on 23 Nov 2007 09:28 UTC**
BTW majority of the _doc strings is translateable now.
The related thing which bothers me currently is ability to have list of commands within the given appmodule built dynamically.
I have started working on this but the problem I am facing is I haven't got an idea how to report keyboard shortcuts and make them translateable as well.
E.G. currently it gives me clear	review current word. I think we would rather like to have numpad5 listed.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 23 Nov 2007 14:04 UTC**
Replying to [pvagner](comment:3):
> BTW majority of the _doc strings is translateable now.
Can you think of any areas where the docstrings haven't been translated? If not, this ticket can probably be closed.

> The related thing which bothers me currently is ability to have list of commands within the given appmodule built dynamically.
I've been thinking about this for a while, but haven't come up with a nice way to order the scripts. Alphabetical order is not particularly useful in some cases. Also, they should probably be categorised. In any case, I think this is probably an issue for another ticket.

**Comment by michaelDCurran** [comment]
**Comment by pvagner on 24 Nov 2007 20:59 UTC**
Replying to [jteh](comment:4):
> Can you think of any areas where the docstrings haven't been translated? If not, this ticket can probably be closed.
> 
Yes I can find no more....
> I've been thinking about this for a while, but haven't come up with a nice way to order the scripts. Alphabetical order is not particularly useful in some cases. Also, they should probably be categorised. In any case, I think this is probably an issue for another ticket.
I am thinking about adding name and category fields the same way _description is used currently and then sort by category. The categories I can think of at the moment are:

*caret navigation
*object navigation
*status reporting
*settings and toggle scripts

But sorry, we will perhaps create a seperate thread as you pointed out.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 14 Jan 2008 23:25 UTC**
